db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/DataDictionary.java

```diff
@@ -2319,4 +2320,18 @@ PermDescriptor getGenericPermissions(UUID objectUUID,
     UUID createNewDatabaseAndDatabaseOwner(TransactionController tc, String dbName, UserDescriptor dbOwner) throws StandardException;
 
     void enableMultiDatabase(boolean value) throws StandardException;
+
+    /**
+     * Returns for a given conglomerate the ID of the transaction used to create it
+     * @param conglom the conglomerate ID.
+     * @return The ID of the transaction used to create it.
+     * @throws StandardException In case of e.g. connection error to HBase admin.
+     */
+    long getConglomerateCreationTxId(long conglom) throws StandardException;
+
+    long getTxnAt(long ts) throws StandardException;
+
+    boolean txnWithin(long period, long pastTx) throws StandardException;
+
+    boolean txnWithin(long period, Timestamp pastTx) throws StandardException;
```

Comment:
getTxnAt and txnWithin are not related to database data dictionary, so you may want to refactor it to somewhere else.

Reply:
I changed it, and moved these methods to `TransactionController` instead, please let me know if you think there is a better location for them.